src/Core/Scheduler/Cache.js

```diff
@@ -49,44 +51,68 @@ const Cache = {
      * @name module:Cache.get
      * @function
      *
-     * @param {string} key
+     * @param {string} key1
+     * @param {string} [key2]
+     * @param {string} [key3]
      *
      * @return {Object}
      */
-    get: (key) => {
-        const entry = data.get(key);
-        const stat = stats.get(key) || stats.set(key, { hit: 0, miss: 0 });
-
-        if (entry) {
-            stat.hit++;
-            entry.lastTimeUsed = Date.now();
-            return entry.value;
+    get: (key1, key2, key3) => {
+        if (data[key1] == undefined) {
+            // eslint-disable-next-line
+            return;
+        } else if (data[key1][key2] == undefined) {
+            return data[key1].value;
+        } else if (data[key1][key2][key3] == undefined) {
+            return data[key1][key2].value;
+        } else {
+            return data[key1][key2][key3].value;
         }
-
-        stat.miss++;
     },
 
     /**
-     * Adds or updates an entry with a specified key. A lifetime can be added,
-     * by specifying a numerical value or using the {@link Cache.POLICIES}
-     * values. By default an entry has an infinite lifetime.
+     * Adds or updates an entry with specified keys (up to 3). A lifetime can be
+     * added, by specifying a numerical value or using the {@link
+     * Cache.POLICIES} values. By default an entry has an infinite lifetime.
+     * Caution: it overrides any existing entry already set at this/those key/s.
      *
      * @name module:Cache.set
      * @function
      *
-     * @param {string} key
      * @param {Object} value
-     * @param {number} [lifetime]
+     * @param {number} [lifetime=Infinity]
+     * @param {string} key1
+     * @param {string} [key2]
+     * @param {string} [key3]
      *
      * @return {Object} the added value
      */
-    set: (key, value, lifetime = Infinity) => {
-        const entry = {
+    set: (value, lifetime = Infinity, key1, key2, key3) => {
```

Comment:
could you remove default value for `lifetime`?

Reply:
Done